docs/components/datetimepicker.md

```diff
@@ -0,0 +1,245 @@
+    <KDateTimePicker
+      @changed="newVal => currentValue1 = newVal"
+      defaultMessage="Please select a time range"
+      calendarMode="dateTime"
+      :isRange="true"
+      :timePeriods="exampleTimeFrames"
+      :defaultCustom="defaultRangeWeek"
+      v-model="currentValue1"
+    />
+    <div class="mt-6">Emitted value: <pre>{{ JSON.stringify(currentValue1) }}</pre></div>
+  </template>
+</KCard>
+
+```html
+<KDateTimePicker 
+  @change="newVal => currentValue = newVal"
+  calendarMode="dateTime"
+  :defaultCustom="defaultRangeWeek"
+  :timePeriods=[...]
+/>
+```
+
+### Relative - show only time frame buttons
+
+Passing in a selected relative time frame value will default the component to default to the "Relative" tab.
+
+<KCard>
+  <template v-slot:body>
+    <KDateTimePicker
+      @changed="newVal => currentValue2 = newVal"
+      defaultMessage="Please select a time range"
+      :timePeriods="exampleTimeFrames"
+      :defaultRelative="defaultRelativeTimeframe"
+      v-model="currentValue2"
+    />
+    <div class="mt-6">Emitted value: <pre>{{ JSON.stringify(currentValue2) }}</pre></div>
+  </template>
+</KCard>
+
+```html
+<KDateTimePicker
+  @changed="newVal => currentValue = newVal"
+  defaultMessage="Please select one of the time ranges below"
+  :timePeriods="exampleTimeFrames"
+  :defaultRelative="defaultRelativeTimeframe"
+/>
+```
+
+### Custom ranges - show only the calendar
+
+Calendar Display only - when the `timePeriods` prop is omitted, only the **Custom** tab is shown.
+
+<KCard>
+  <template v-slot:body>
+    <KDateTimePicker
+      @changed="newVal => currentValue3 = newVal"
+      defaultMessage="Please select some dates"
+      calendarMode="date"
+      :defaultCustom="defaultRangeTwoDay"
+      v-model="currentValue3"
+    />
+    <div class="mt-6">Emitted value: <pre>{{ JSON.stringify(currentValue3) }}</pre></div>
+  </template>
+</KCard>
+
+```html
+<KDateTimePicker
+  @changed="newVal => currentValue = newVal"
+  defaultMessage="Please select some dates"
+  :defaultCustom="{
+    start: new Date().getTime(),
+    end: new Date().getTime() - (2*24*60*60*1000)
+  }"
+/>
+```
+
+## Props
+
+### calendarMode
+
+- Allowed values are: `date`, `time`, `dateTime`
+
+### isRange
+
+Along with the `calendarMode` prop, this determines the whether the calendar allows selection of a single date or time, as opposed to a range of start and end values.
+
+### timePeriods
+
+An array of time frame values to be displayed as buttons in the "Relative" section of the popover, eg:
+
+```json
+[
+  {
+    section: 'Last',
+    values: [
+      { timeframeText: '15 minutes', start: function(), end: function() },
+      { timeframeText: '12 hours', start: function(), end: function() },
+      { timeframeText: '24 hours', start: function(), end: function() }
+    ]
+  },
+  {
+    section: 'Previous',
+    values: [
+      { timeframeText: 'Week', start: function(), end: function() }
+    ]
+  }
+]
+```
+
+### value
```

Comment:
question: does the `value` prop get mapped to the `v-model`? 

If yes, let's go ahead and rename the prop to `modelValue` to match the Vue 3 spec so we don't have to do it later.